flocker/control/httpapi.py

```diff
@@ -134,6 +136,32 @@ def render_if_matches(self, request, **route_arguments):
     return render_if_matches
 
 
+@lru_cache(1)
+def _extract_containers_state(deployment_state):
+    """
+    Turn the deployment state into serializable simple types for every
+    container in the deployment state.
+
+    N.B.: Callers of this function should not mutate the returned value as this
```

Comment:
Opened up FLOC-4345 to make this better, preliminary attempts to make the issue go away proved to be not successful.